### PR TITLE
feat(wakatime): added Wakatime segment

### PR DESCRIPTION
**Pull request by Khaos66**
### Prerequisites

- [x] I have read and understood the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes/features)

### Description

This adds a new segment for [wakatime](https://wakatime.com/). It displays the tracked time of the current day

### CODE REVIEW

**Comment by Khaos66** [comment]
Sorry, I forgot to change some comments when copying owm... but it should be all clear now

File: src/segment_wakatime.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	Minutes      int
+	MinutesTotal int
+}
+
+const (
+	// CacheKeyResponse key used when caching the response
+	WTCacheKeyResponse string = "wt_response"
+)
+
+type wtTotals struct {
```
Comment (JanDeDobbeleer) [inline src/segment_wakatime.go:42]: I would expose this as well, no? We can use inheritance to automaticaly add these after parsing the response.

File: src/segment_wakatime.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			if err != nil {
+				return nil, err
+			}
+			return response, nil
+		}
+	}
+
+	apikey := w.props.getString(APIKey, ".")
+	httpTimeout := w.props.getInt(HTTPTimeout, DefaultHTTPTimeout)
+	url := fmt.Sprintf("https://wakatime.com/api/v1/users/current/summaries?start=today&end=today&api_key=%s", apikey)
```
Comment (JanDeDobbeleer) [inline src/segment_wakatime.go]: We could also expose the url including the `api_key` like the `nightscout` segment?

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]
Two remarks, really well done.

File: src/segment_wakatime.go (hunk 1 above)
Comment (Khaos66) [reply src/segment_wakatime.go:42]: Not totally sure what you mean by inheritance, but i cound expose this, yes. Will do and we'll see if that's hat you expected ;)

File: src/segment_wakatime.go (hunk 2 above)
Comment (Khaos66) [reply src/segment_wakatime.go]: I don't see the point in doing so, but sure I will

Comment (JanDeDobbeleer) [reply src/segment_wakatime.go]: That's a question more than a remark. I was wondering if people might want to use different time spans (hardcoded to a day now).

File: src/segment_wakatime.go (hunk 1 above)
Comment (JanDeDobbeleer) [reply src/segment_wakatime.go:42]: By adding that struct to the segment struct. You can then also parse the response with that embedded struct (like with git and scm).

**Comment by Khaos66** [comment]
@JanDeDobbeleer should be good now

**Comment by Khaos66** [comment]
hmm.. I crashed the linter with my comment... again ... will fix

**Comment by Khaos66** [comment]
@JanDeDobbeleer can you plz help me? Whats the problem of the code validator now?
Edit: 
I found the problem

**Comment by JanDeDobbeleer** [comment]
I'm reviewing this in the morning, need my desk for this. I think I made things worse by not being clear enough, I'll fix that together with you 😅

**Comment by Khaos66** [comment]
> I'm reviewing this in the morning, need my desk for this. I think I made things worse by not being clear enough, I'll fix that together with you 😅

Ok. Feels strange being on the other side of this for once. In my day to day business I'm the one giving ~_unclear_~ **high level** advice 😅 
Take your time. We'll get there

**Comment by Khaos66** [comment]
@JanDeDobbeleer It's feature complete now. Please elaborate the inheritance issue. Then we can merge I guess =) 
You don't need to hurry. Take your time.

**Comment by JanDeDobbeleer** [comment]
@Khaos66 I'm writing an implementation based on your commits. I had what seems to be a brilliant idea.

**Comment by Khaos66** [comment]
Great news. I can't wait to see it 😃

**Pull request closed without merging**